vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3205,13 +3208,56 @@ public void setMultiSort(boolean multiSort) {
      * @see MultiSortPriority
      */
     public void setMultiSort(boolean multiSort, MultiSortPriority priority) {
-        Objects.requireNonNull(priority,
-                "Multi-sort priority must not be null");
         setMultiSort(multiSort);
         updateMultiSortPriority(priority);
     }
 
+    /**
+     * Sets whether multiple column sorting is enabled on the client-side.
+     *
+     * @param multiSort
+     *            {@code true} to enable sorting of multiple columns on the
+     *            client-side, {@code false} to disable
+     * @param onShiftClickOnly
+     *            {@code true} to enable multi-sort by shift-clicking (when
+     *            {@code multiSort = true}), {@code false} for normal multi-sort
+     *            behavior
+     */
+    public void setMultiSort(boolean multiSort, boolean onShiftClickOnly) {
+        setMultiSort(multiSort);
+        if (multiSort) {
+            updateMultiSortShiftClickOnly(onShiftClickOnly);
+        }
+    }
+
+    /**
+     * Sets whether multiple column sorting is enabled on the client-side.
+     *
+     * @param multiSort
+     *            {@code true} to enable sorting of multiple columns on the
+     *            client-side, {@code false} to disable
+     * @param priority
+     *            the multi-sort priority to set, not {@code null}
+     * @param onShiftClickOnly
+     *            {@code true} to enable multi-sort by shift-clicking (when
+     *            {@code multiSort = true}), {@code false} for normal multi-sort
+     *            behavior
+     *
+     * @see MultiSortPriority
+     */
+    public void setMultiSort(boolean multiSort, MultiSortPriority priority,
+            boolean onShiftClickOnly) {
+        setMultiSort(multiSort, onShiftClickOnly);
+        updateMultiSortPriority(priority);
+    }
+
+    private void updateMultiSortShiftClickOnly(Boolean onShiftClickOnly) {
```

Comment:
Would rename this as `updateMultiSortOnShiftClick` (also the param name) since that's effectively the property it updates

Reply:
Done.

Reply:
No, I meant only the private method that just sets the property. The public API does more than that so it should not be renamed.

Reply:
Looking at the diff, I noticed it's missing the `On` word in the middle. So it could be `grid.isMultiSortOnShiftClickOnly`.

Reply:
Or should we drop the getter? I didn't see it mentioned in the API design at https://github.com/vaadin/platform/issues/3378